lef21/src/read.rs

```diff
@@ -201,12 +204,25 @@ impl<'src> LefLexer<'src> {
         let tok = self.emit(TokenType::WhiteSpace);
         Ok(Some(tok))
     }
-    /// Lex a number
-    fn lex_number(&mut self) -> LefResult<Option<Token>> {
-        while self.accept(|c| c.is_digit(10) || c == '.') {
+    /// Lex a number, but if not truly a number, return token with ttype [TokenType::Name]
+    fn lex_number(&mut self, leadchar: char) -> LefResult<Option<Token>> {
+        let mut nstring = String::new();
+        nstring.push(leadchar);
```

Comment:
The numeric conversion routines are what they are but I am still a bit annoyed by the need to allocate a new String, prepend the the peeked char and then tack on a str slice  from `self.chars.as_str()`.
For some reason that doesnt make sense to me the chars iterator is advanced one position so that I couldn't get just grab a simple slice of it (without missing the necessary first character).